py/kubeflow/testing/cleanup_ci.py

```diff
@@ -293,20 +293,28 @@ def cleanup_firewall_rules(args):
   logging.info("expired firewall rules:\n%s", "\n".join(expired))
 
 def cleanup_instance_groups(args):
+  logging.info("Cleanup instance groups")
+
   if not args.gc_backend_services:
     return
-
   credentials = GoogleCredentials.get_application_default()
   compute = discovery.build('compute', 'v1', credentials=credentials)
   instanceGroups = compute.instanceGroups()
   next_page_token = None
-  expired = []
+  deleted = []
   unexpired = []
   in_use = []
 
-  # TODO(jlewi): We should check whether the instance group is in use
-  # before deleting it. At least in pantheon it looks like instance groups
-  # are listed as in use by clusters.
+  auto_deploy_patterns = re.compile("gke-kf-vmaster.*")
+  e2e_pattern = re.compile("k8s-ig-.*")
+
+  def ig_name_to_infra_type(name):
```

Comment:
Why not just modify name_to_infra to return E2E_OWNERLESS if the name matches various regex and then define OWNERLESS_PATTERNS that includes "k8s-ig.*"

Reply:
changed.